fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/LuceneSearchProvider.java

```diff
@@ -317,6 +325,96 @@ private void updateDimensionRow(
         writer.updateDocument(keyTerm, luceneDimensionRowDoc);
     }
 
+    @Override
+    public void replaceIndex(String newLuceneIndexPathString) {
+        lock.writeLock().lock();
+        try {
+            Path oldLuceneIndexPath = Paths.get(luceneIndexPath);
+            String tempDir = oldLuceneIndexPath.resolveSibling(oldLuceneIndexPath.getFileName() + "_old").toString();
+
+            LOG.debug("Moving old Lucene index directory from {} to {} ...", luceneIndexPath, tempDir);
+            moveDirEntries(luceneIndexPath, tempDir);
```

Comment:
Is there any logic to prevent an unexpected abort during the process moving old index to temp dir? It could result in an incomplete index. Would it be better to use a copy/delete fashion rather than move?

Reply:
`replaceIndex` itself is in copy/delete fashion.